clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/DcosConfiguration.groovy

```diff
@@ -76,20 +75,24 @@ class DcosConfiguration {
     new PollingDcosDeploymentMonitor(operationPoller)
   }
 
-  private static Optional<DCOSAuthCredentials> buildDCOSAuthCredentials(String accountName, String uid, String password, String serviceKey) {
-    Optional<DCOSAuthCredentials> dcosAuthCredentials = Optional.empty()
-
-    if (uid && serviceKey && password) {
-      LOGGER.warn("Ignoring account [${accountName}]: A password and serviceKey was given for the account making it invalid.")
-    } else if (uid && password) {
-      dcosAuthCredentials = Optional.of(DCOSAuthCredentials.forUserAccount(uid, password))
-    } else if (uid && serviceKey) {
-      dcosAuthCredentials = Optional.of(DCOSAuthCredentials.forServiceAccount(uid, serviceKey))
-    } else {
-      LOGGER.warn("Ignoring account [${accountName}]: Neither a uid/password or a uid/serviceKey combination was provided.")
+  private static Config buildConfig(final Account account) {
+    Config.builder().withCredentials(buildDCOSAuthCredentials(account))
+      .withInsecureSkipTlsVerify(account.insecureSkipTlsVerify)
+      .withCaCertData(account.caCertData)
+      .withCaCertFile(account.caCertFile).build()
+  }
+
+  private static DCOSAuthCredentials buildDCOSAuthCredentials(Account account) {
+    DCOSAuthCredentials dcosAuthCredentials = null
+
+    // If they've specified both for whatever reason, might as well go with one of them.
```

Comment:
I'd prefer to still consider that an error.  If we assume it should use the password and it's actually a service account then the user would still get an error but it might be confusing to see just that the authentication failed.  Even if we happen to pick the right one, it seems quite possible that this was a configuration error and the account that works might not even be the one that was intended to be configured.